src/poll.rs

```diff
@@ -418,6 +413,19 @@ impl Poll {
     }
 }
 
+impl Deref for Poll {
```

Comment:
I prefer to reserve `Deref` implementations for "smart pointer" types. I would opt to keep a `fn registry(&self) -> `&Registry` function.

IMO, a `Deref` impl is a pretty heavy guarantee to maintain for forwards compat.

Reply:
I do like that we can do `Poll::register` with this but I'll change it back.

Reply:
You can always add `register`, `reregister`, `deregister` fns directly on `Poll`.

Reply:
Let's merge this as is and we can always add it in the future.